src/modules/blockchainApplication/explore/components/BlockchainApplicationList/BlockchainApplicationList.test.js

```diff
@@ -1,39 +1,46 @@
-import React from 'react';
-import { MemoryRouter } from 'react-router';
 import { fireEvent, screen } from '@testing-library/react';
-import mockBlockchainApplications from '@tests/fixtures/blockchainApplicationsExplore';
 import { usePinBlockchainApplication } from '@blockchainApplication/manage/hooks/usePinBlockchainApplication';
-import { renderWithRouter } from 'src/utils/testHelpers';
+import {
+  renderWithRouterAndQueryClient,
+  rerenderWithRouterAndQueryClient,
+} from 'src/utils/testHelpers';
+import { useBlockchainApplicationExplore } from '../../hooks/queries/useBlockchainApplicationExplore';
+import { mockBlockchainApp } from '../../__fixtures__';
 import BlockchainApplicationList from './BlockchainApplicationList';
-import { BLOCKCHAIN_APPLICATION_LIST_LIMIT } from '../../const/constants';
+
+const mockTogglePin = jest.fn();
+const mockedPins = [mockBlockchainApp.data[0].chainID];
+const mockApplyFilters = jest.fn();
+const mockFetchNextPage = jest.fn();
 
 jest.useFakeTimers();
 jest.mock('@blockchainApplication/manage/hooks/usePinBlockchainApplication');
-const mockTogglePin = jest.fn();
-const mockedPins = [mockBlockchainApplications[0].chainID];
+jest.mock('../../hooks/queries/useBlockchainApplicationExplore');
+jest.mock('@common/hooks', () => ({
+  ...jest.requireActual('@common/hooks'),
+  useFilter: jest.fn(() => ({
+    applyFilters: mockApplyFilters,
+  })),
+}));
```

Comment:
please take a look to docs, https://jestjs.io/docs/jest-object#jestrequireactualmodulename
```suggestion
jest.mock('@common/hooks', () => ({
  __esModule: true,
  ...jest.requireActual('@common/hooks'),
  useFilter: jest.fn(() => ({
    applyFilters: mockApplyFilters,
  })),
}));
```

Reply:
by the way why are your mocking it please use jest.spyOn()

Reply:
@soroushm Resolved